README.md

```diff
@@ -108,24 +131,85 @@ if($player instanceof Player && $player->isOnline()){
+This allows the camera to visually track a specific entity.  
+Unlike SetActorLinkPacket, it does not follow the entity automatically.  
+
+<details align="center">
+	<summary>See demo</summary>
+
+https://github.com/user-attachments/assets/38cd6bf1-f666-4635-870b-2d51b12bfa3f
+
+</details>
+
 ```php
-use muqsit\libcamera\libcamera;
+use pocketmine\event\player\PlayerInteractEvent;
 use muqsit\libcamera\CameraInstruction;
-use pocketmine\network\mcpe\protocol\types\camera\CameraTargetInstruction;
+use pocketmine\entity\Zombie;
+use muqsit\libcamera\libcamera;
+use pocketmine\network\mcpe\protocol\types\camera\CameraSetInstructionEase;
+use pocketmine\network\mcpe\protocol\types\camera\CameraSetInstructionEaseType;
+use pocketmine\network\mcpe\protocol\types\camera\CameraSetInstructionRotation;
 use pocketmine\math\Vector3;
-use pocketmine\player\Player;
-
+use muqsit\libcamera\CameraPresetRegistry;
 // ...
-if($player instanceof Player && $player->isOnline()){
-	/** 
-	 * @phpstan-param Vector3|null $targetCenterOffset
-	 * @phpstan-param int $actorUniqueId
-	 */
-	CameraInstruction::target(
-		targetCenterOffset: Vector3::zero(), // no offset
-		actorUniqueId: $player->getId() // for example target the player
-	)->send($player);
-}
+
+	/** @var array<string, true> */
+	private $set = [];
+
+	public function ina(PlayerInteractEvent $event) : void{
+		$player = $event->getPlayer();
+		if(!$player->isSneaking()){
+			return;
+		}
+
+		//Removes camera tracking. Note that target and free cameras are managed separately.
+		if(isset($this->set[$player->getName()])){
+			CameraInstruction::removeTarget()->send($player);
+			CameraInstruction::clear()->send($player);
+			unset($this->set[$player->getName()]);
+			return;
+		}
+
+		//Find the most different zombie entities
+		$nearest = null;
+		$nearestDistance = PHP_INT_MAX;
+		foreach($player->getWorld()->getEntities() as $entity){
+			if($entity instanceof Zombie){
+				$distance = $player->getPosition()->distance($entity->getPosition());
+				if($nearestDistance >= $distance){
+					$nearest = $entity;
+					$nearestDistance = $distance;
+				}
+			}
+		}
+
+		if($nearest === null){
+			$player->sendMessage("No Zombie");
+			return;
+		}
+
+		//
+		CameraInstruction::set(
+			preset: CameraPresetRegistry::TARGET(),
+			ease: null,
+			camera_pos: $player->getPosition()->add(0, $player->getEyeHeight(), 0),
+			rot: new CameraSetInstructionRotation(
+				(float) $player->getLocation()->getPitch(), //pitch
+				(float) $player->getLocation()->getYaw() //yaw
+			),
+			facing_pos: null
+		)->send($player);
+
+		//To use CameraInstruction::target you first need to make it a free camera.
+		CameraInstruction::target(
+			targetCenterOffset: Vector3::zero(), // no offset
+			actorUniqueId: $nearest->getId() // for example target the player
+		)->send($player);
+
+		//Manages which packets have been sent
+		$this->set[$player->getName()] = true;
+	}
```

Comment:
Not only is this absolute overkill, you also seem to have misunderstood how the instructions send the packets - this, as far as i remember, is automatically handled on send

Reply:
I would need to unlock it with a sneak click, start it with a normal click, or unlock it over time